.gitmessage

```diff
@@ -0,0 +1,9 @@
+# If this commit is applied, it will ....
```

Comment:
I would also like to encourage the use of short prefixes in commit sets, and make it an explicit requirement.
there's a lot of stuff in the logs like this:
10e6802114 Leadwork heavy duty barrel sipersion to 90
5a91b14a26 fromatting
9d2798ec35 formatting more
f67f9a5eed formatting json
7464d0afdd styling
66ca50acea Pistols and shotgun
4154ddb42d Rollback some changes
66944f5ddc Removed excess brackets

and it's hard to tell if those are related changes or not.
something like
10e6802114 leadwork rifles rework: Leadwork heavy duty barrel sipersion to 90
5a91b14a26 leadwork rifles rework: fromatting
9d2798ec35 leadwork rifles rework: formatting more
f67f9a5eed leadwork rifles rework: formatting json
7464d0afdd leadwork rifles rework: styling
66ca50acea leadwork rifles rework: Pistols and shotgun
4154ddb42d leadwork rifles rework: Rollback some changes
66944f5ddc leadwork rifles rework: Removed excess brackets

makes reviewing logs to see if you should look at a particular commit more closely a lot easier.

Reply:
I generally agree with You that some commits are poorly commented. In some cases `formatting json` might be acceptable because it does fully and completely describes commit but adding long (22 characters in Your example) prefixes doesn't always bring any additional value.

I would encourage better fleshed out commit titles  with optional later description and/or linkage to Issues.

Reply:
"formatting json" tells me that JSON is being formatted, but there 100s of JSON files and I'd like to have some hints as to which ones. The prefix may be wordy but by looking at it, I can guess that data/json/mapgen/pond.json probably wasn't involved.

Most commit summaries are short, 10-30 characters. Adding a 22 character prefix is not going to push most of them over 50 characters, and there's no reason to not support 70 characters in the review summary line as far as I know.

Reply:
Everything about this PR is great except the "Maximum of 50 Characters" line, even your references say:
Summarize changes in *around* 50 characters or less
50 characters is not a hard limit, just a rule of thumb.
So shoot for 50 characters, but consider 72 the hard limit.

So I'm going to edit it to reflect the recommendation and the hard limit.

Side note, I'm not sure how functional having category labels within commit titles is, especially considering that compliance would hover at nearly zero.